IPython/core/magics/execution.py

```diff
@@ -115,6 +108,34 @@ def visit_For(self, node):
         return node
 
 
+class Timer(timeit.Timer):
+    """Timer class that explicitly uses self.inner
+    
+    which is an undocumented implementation detail of CPython,
+    not shared by PyPy.
+    """
+    # Timer.timeit copied from CPython 3.4.2
+    def timeit(self, number=timeit.default_number):
+        """Time 'number' executions of the main statement.
+
+        To be precise, this executes the setup statement once, and
+        then returns the time it takes to execute the main statement
+        a number of times, as a float measured in seconds.  The
+        argument is the number of times through the loop, defaulting
+        to one million.  The main statement, the setup statement and
+        the timer function to be used are passed to the constructor.
+        """
+        it = itertools.repeat(None, number)
+        gcold = gc.isenabled()
+        gc.disable()
+        try:
+            timing = self.inner(it, self.timer)
```

Comment:
CPython sets it in `__init__`, we override it in `%timeit`, it's used in `.timeit`. This patch changes nothing at all for CPython. PyPy doesn't set it in `__init__`, we set it after, and this class ensures that it's used in `.timeit`.